dashboard/app/dsl/user_id_replace_dsl.rb

```diff
@@ -0,0 +1,2 @@
+class UserIdReplaceDSL < ExternalDSL
+end
```

Comment:
#13833 was a different enough level type that I was okay with creating a new type - in this case we want exactly the behavior of External, with this one extra little thing (replacing user_id).